### PR TITLE
feat(PE-6613): update manage domains

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/components/pages/Undernames/Undernames.tsx [hunk 1]
```diff
@@ -45,6 +50,11 @@ function Undernames() {
         ),
       );
     }
+
+    setisMaxUndernameCount(
```
Comment (thiagocarvalhodev) [inline src/components/pages/Undernames/Undernames.tsx:54]:
```suggestion
    setIsMaxUndernameCount(
```

File: src/utils/searchUtils/searchUtils.ts [hunk 2]
```diff
@@ -191,3 +192,12 @@ export function isValidGateway(gateway: string) {
 export function isValidURL(url: string) {
   return url ? URL_REGEX.test(url) : false;
 }
+
+export function isMaxLeaseDuration(duration: number | string) {
+  return (
+    (duration &&
+      typeof duration === 'number' &&
+      duration >= MAX_LEASE_DURATION) ||
+    duration === 'Indefinite'
```
Comment (thiagocarvalhodev) [inline src/utils/searchUtils/searchUtils.ts:201]: We use the same String on `src/components/forms/DomainSettings/DomainSettings.tsx`. Consider creating a new const

File: src/components/modals/ant-management/UpgradeAntsModal/UpgradeAntsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        }).catch(() => {
+          failedUpgrades.push(antId);
+          eventEmitter.emit('error', {
+            name: 'Upgrade Error',
+            message: `Issue upgrading ANT ${antId}, please try again later`,
+          });
+        });
+        setProgress((prev) => Math.round(prev + 100 / antIds.length));
+      }
+      if (failedUpgrades.length < antIds.length) {
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/components/modals/ant-management/UpgradeAntsModal/UpgradeAntsModal.tsx:118]: Not on this one no. Its just not scaleable - they can sort by source code id to see that though.

File: src/components/forms/DomainSettings/DomainSettings.tsx [hunk 4]
```diff
@@ -88,7 +105,7 @@ function DomainSettings({
 
       if (data?.processId) {
         queryClient.invalidateQueries({
-          queryKey: ['ant', data.processId.toString()],
+          queryKey: ['ant', data?.processId.toString()],
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/components/forms/DomainSettings/DomainSettings.tsx:108]: Note the check in the if statement

File: src/components/modals/ant-management/UpgradeAntsModal/UpgradeAntsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  }, [luaCodeTx, ants]);
+
+  function handleClose() {
+    setVisible(false);
+    setAccepted(false);
+    setProgress(-1);
+  }
+
+  async function upgradeAnts() {
+    if (progress > -1) return;
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/components/modals/ant-management/UpgradeAntsModal/UpgradeAntsModal.tsx:82]: Added a useCallback for isUpdatingAnts

**Comment by atticusofsparta** [comment]
@thiagocarvalhodev  i couldnt replicate the max lease duration thing you had shown, but thats what it looks like when loading IIRC which is out of the scope of this pr, though is an issue. Will create a ticket to address.

**Review by thiagocarvalhodev — DISMISSED** [review]
LGTM

**Comment by thiagocarvalhodev** [comment]
> @thiagocarvalhodev i couldnt replicate the max lease duration thing you had shown, but thats what it looks like when loading IIRC which is out of the scope of this pr, though is an issue. Will create a ticket to address.

Sounds good! Thanks

File: src/components/forms/DomainSettings/SourceCodeIdRow.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+              'N/A'
+            )}
+          </span>
+        }
+        action={
+          // editable &&
+          // sourceCodeTxId &&
+          // ![ANT_LUA_ID, data?.originalTxId]
+          //   .filter((id) => id !== undefined)
+          //   .includes(sourceCodeTxId)
```
Comment (thiagocarvalhodev) [inline src/components/forms/DomainSettings/SourceCodeIdRow.tsx:53]: Remove this commented code

File: src/components/forms/DomainSettings/SourceCodeIdRow.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+              'N/A'
+            )}
+          </span>
+        }
+        action={
+          editable &&
+          sourceCodeTxId &&
+          ![ANT_LUA_ID, data?.originalTxId]
+            .filter((id) => id !== undefined)
+            .includes(sourceCodeTxId) && [
```
Comment (thiagocarvalhodev) [inline src/components/forms/DomainSettings/SourceCodeIdRow.tsx:53]: It's not implicit what this means. Can you add this to a variable with a meaningful name in this context?

**Review by thiagocarvalhodev — DISMISSED** [review]
Just nits. LGTM.

**Review by thiagocarvalhodev — DISMISSED** [review]

**Pull request merged**